les_11_1/SnakeGame.py

```diff
@@ -78,11 +78,24 @@ def update(self, events, dt):
             self.snake = self.snake.take(self.length)
 
             self.cooldown = self.speed
-
+            
+            if self.isdead(self.snake):
+                print("you ded")
+                exit()
+            
             if self.snake.pos.is_same(self.food):
                 self.length += 1
                 snake = Snake(self.food, self.snake)
                 self.setfood()
+    
+    def isdead(self, snake):
+        headpos = snake.pos
+        snake = snake.tail
+        while snake != None:
```

Comment:
It is better to use `is not` see pep8

[pep-0008 programming recommudations](https://www.python.org/dev/peps/pep-0008/#programming-recommendations)

> Comparisons to singletons like None should always be done with is or is not , never the equality operators.
> 
> Also, beware of writing if x when you really mean if x is not None -- e.g. when testing whether a variable or argument that defaults to None was set to some other value. The other value might have a type (such as a container) that could be false in a boolean context!